### ISSUE TITLE
Remotes that this works with

### ISSUE THREAD

**Issue by s00500** [fsievers22/esphome-ble-remote#10]
Hi!
I tried this with 3 different generations of remotes:
The one without the alexa mic, the first one with the mic and the first one that had IR on board

Only the one that has IR I could get to work....

Do you have any additional infos here ? Could you make others work ?

(Minor info: I also got the crash issue reported elsewhere, I did downgrade to using the arduino framework to avoid it for now)

Greetings,
Lukas

**Comment by Bretthansenpc** [comment]
what do you mean by downgraded? I'm still having this issue

**Comment by s00500** [comment]
I have checked out 9d8788c652682b5879bed466e4f50b00d098a84e and then adjusted the mac address in the header file

**Comment by Bretthansenpc** [comment]
sorry im confused my coding is amateur at best haha. Is this where I can change the mac? And how would I write it? haha i can help with your other questions I think once I get this working haha

`BleHidClientComponent(const char ble_address[]);`

**Comment by s00500** [comment]
checkout the repo at https://github.com/fsievers22/esphome-ble-remote/tree/9d8788c652682b5879bed466e4f50b00d098a84e

Then change the SERVER_ADDRESS that is hardcoded into the .h file

Notice that I am also not using this as a ESPHomecomponent, but for a different project....

As for the remotes it seems that only a very certain generation of them supports ble hid.... and also there are different firmware versions that hold the info about the key in different bytes....

**Comment by fsievers22** [comment]
So I think I have fixed the issue of crashing on  keypress in PR #11. I can't say which remotes work with the code. I have used it with the remote with volume control buttons but no blue alexa button and no netflix/prime video buttons. I also used the code with an nvidia shield remote, but I had to change the keymap here: https://github.com/fsievers22/esphome-ble-remote/blob/3176d4b0d96fbc95f881bd82fe875f8dad0aa3dd/ble_hid_client.cpp#L41-L44
